pkg/machines/components/networks/utils.js

```diff
@@ -229,3 +229,34 @@ export function isIpv6InNetwork(network, prefix, ip) {
 
     return network == ip;
 }
+
+/* If you take an IP address an apply the network mask (bitwise and) then you get the network address - no host can have such address
+ * This function returns the first usable address of the block
+ * @param {string} ipv4
+ * @param {string} netmask
+ */
+export function ipv4FirstUsableAddress(ipv4, netmask) {
+    // Within a subnet, two host addresses - all-zeros and one all-ones are reserved as network address and broadcast, respectively.
+    if (!ipv4)
+        return undefined;
+
+    let mask = netmask;
+    if (netmask.split(".").length == 4)
+        mask = netmaskToCIDR(netmask);
+    // For a /31 subnet the number of usable addresses is zero.
+    if (mask > 30)
+        return undefined;
+
+    return ipv4.split(".")
+            .map((item, index) => (item & netmask.split(".")[index]) + Number(index == 3))
```

Comment:
This looks overcomplicated and error prone. At this point you should already have made sure that it's not an invalid IP address, either by existing earlier validation, or by doing additional validation yourself. It's enough to split at the last dot and increase that, or keep doing what you do, and just change the `[3]` element. Also, you need to handle the case that the last component is 254, as 255 is not a valid IP (it's the broadcast).

What is that IP exactly, can it really be arbitrary? In the issue the screenshot shows ".0", which isn't an IP, but just gives a network. Will it always be 0? Can it be some actual IP? Do the two cases really mean the same?